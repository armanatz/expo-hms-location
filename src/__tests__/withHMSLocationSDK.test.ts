import {
  _applyBouncyCastleResolutionFix,
  _addHMSImplementation,
  _applyHMSAGConnectServicesPlugin,
} from '../actions/withHMSLocationSDK';

const appBuildGradle = `apply plugin: "com.facebook.react"

android {
}

dependencies {
}`;

describe('app build.gradle modifications', () => {
  it('applies agcp plugin', () => {
    // Test that it adds the plugin correctly
    const firstResult = _applyHMSAGConnectServicesPlugin(appBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didMerge).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the plugin
    const secondResult = _applyHMSAGConnectServicesPlugin(firstResult.contents);
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didMerge).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });

  it('adds hms implementation to dependencies', () => {
    // Test that it adds the implementation correctly
    const firstResult = _addHMSImplementation(appBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didMerge).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the implementation
    const secondResult = _addHMSImplementation(firstResult.contents);
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didMerge).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });

  it('applies bouncy castle resolution fix', () => {
    // Test that it applies the fix correctly
    const firstResult = _applyBouncyCastleResolutionFix(appBuildGradle);
    expect(firstResult.contents).toMatchSnapshot();
    expect(firstResult.didMerge).toBe(true);
    expect(firstResult.didClear).toBe(false);

    // Test that it doesn't duplicate the fix
    const secondResult = _applyBouncyCastleResolutionFix(firstResult.contents);
    expect(secondResult.contents).toBe(firstResult.contents);
    expect(secondResult.didMerge).toBe(false);
    expect(secondResult.didClear).toBe(false);
  });
});
