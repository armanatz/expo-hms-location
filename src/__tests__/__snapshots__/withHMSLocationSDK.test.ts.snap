// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app build.gradle modifications adds hms implementation to dependencies 1`] = `
"apply plugin: "com.facebook.react"

android {
}

dependencies {
    // @generated begin @armanatz/expo-hms-location-implementation - expo prebuild (DO NOT MODIFY) sync-8eee6c6552e2dedbd04423ac3fcff0d218ec93bb
    implementation("com.huawei.hms:location:6.12.0.300")
    // @generated end @armanatz/expo-hms-location-implementation
}"
`;

exports[`app build.gradle modifications applies agcp plugin 1`] = `
"apply plugin: "com.facebook.react"
// @generated begin @armanatz/expo-hms-location-agconnect - expo prebuild (DO NOT MODIFY) sync-1c63e1459bc83532e84404ddae98c345ce2ce150
apply plugin: "com.huawei.agconnect"
// @generated end @armanatz/expo-hms-location-agconnect

android {
}

dependencies {
}"
`;

exports[`app build.gradle modifications applies bouncy castle resolution fix 1`] = `
"apply plugin: "com.facebook.react"

android {
    // @generated begin @armanatz/expo-hms-location-bouncycastle-fix - expo prebuild (DO NOT MODIFY) sync-59c55b2624e7cca5387fdb71180ab76058a62315
    configurations.all {
        c -> c.resolutionStrategy.eachDependency {
            DependencyResolveDetails dependency ->
                if (dependency.requested.group == 'org.bouncycastle') {
                    dependency.useTarget 'org.bouncycastle:bcprov-jdk15to18:1.70'
                }
        }
    }
    // @generated end @armanatz/expo-hms-location-bouncycastle-fix
}

dependencies {
}"
`;
